spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/autoconfig/GcpCloudSqlAutoConfiguration.java

```diff
@@ -0,0 +1,127 @@
+import java.util.Collections;
+import java.util.function.Supplier;
+
+import javax.sql.DataSource;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.sqladmin.SQLAdmin;
+import com.google.api.services.sqladmin.model.DatabaseInstance;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.gcp.core.GcpProperties;
+import org.springframework.cloud.gcp.sql.GcpCloudSqlProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * Google Cloud SQL starter.
+ *
+ * <p>It simplifies database configuration in that only an instance and database name are provided,
+ * instead of a JDBC URL. If the instance region isn't specified, this starter attempts to get it
+ * through the Cloud SQL API.
+ *
+ * @author João André Martins
+ */
+@Configuration
+@EnableConfigurationProperties(GcpCloudSqlProperties.class)
+@ConditionalOnClass({GcpProperties.class, GoogleCredential.class})
+@AutoConfigureBefore(DataSourceAutoConfiguration.class)
+public class GcpCloudSqlAutoConfiguration {
+
+	@Autowired
+	private GcpCloudSqlProperties gcpCloudSqlProperties;
+
+	@Autowired
+	private GcpProperties gcpProperties;
+
+	@Bean
+	@ConditionalOnMissingBean(DataSource.class)
+	public DataSource cloudSqlDataSource(SQLAdmin sqlAdmin)
+			throws GeneralSecurityException, IOException {
+		Assert.hasText(this.gcpCloudSqlProperties.getDatabaseName(),
+				"spring.cloud.gcp.sql.databaseName name can't be empty.");
+		Assert.hasText(this.gcpCloudSqlProperties.getInstanceName(),
+				"spring.cloud.gcp.sql.instanceName name can't be empty.");
+		Assert.hasText(this.gcpCloudSqlProperties.getUserName(),
+				"spring.cloud.gcp.sql.userName name can't be null.");
+
+		// Look for the region if the user didn't specify one.
+		String region = this.gcpCloudSqlProperties.getRegion();
+		if (!StringUtils.hasText(region)) {
+			DatabaseInstance instance =	sqlAdmin.instances().get(
+					this.gcpProperties.getProjectId(),
+					this.gcpCloudSqlProperties.getInstanceName()).execute();
+
+			region = instance.getRegion();
```

Comment:
Why not combine with the previous line and avoid using a local var for `instance`?